### ISSUE TITLE
Rust file is not recognised when its name contains more then one underscore 

### ISSUE THREAD

**Issue by dthoma** [intellij-rust/intellij-rust#729]
When a rust source file has a name containing at least two underscores the plugin won't recognise it as a rust source file. (I.e. the file does not have the rust icon in the project view and it will open in a plain text editor.)

**Comment by matklad** [comment]
Hm, I can't reproduce this, plugin recognized `__f_o_o__.rs` perfectly. My guess would be that you have something like `__.*` in ignored files:

![ignored](https://cloud.githubusercontent.com/assets/1711539/19312732/6f689258-909c-11e6-9574-09975ebe0d4e.png)

**Comment by dthoma** [comment]
Hm, ok, I've figured it out. Somehow I managed to create an explicit mapping of that file name to a text content type. And this mapping persisted even after deleting and recreating the file or when renaming an other file to that name. I had to remove that mapping from the global idea configuration.
So it is either an Idea bug or just something very strange I must have done at some point in time. Sorry for the trouble.
